src/create-tree.ts

```diff
@@ -62,6 +63,28 @@ export async function createTree(
         result = await value(
           Object.assign(file, { exists: true }) as UpdateFunctionFile
         );
+
+        if (result === deleteFile) {
+          try {
+            // https://developer.github.com/v3/repos/contents/#get-contents
+            await octokit.request("HEAD /repos/{owner}/{repo}/contents/:path", {
+              owner: ownerOrFork,
+              repo,
+              ref: latestCommitSha,
+              path,
+            });
+
+            tree.push({
+              path,
+              mode: "100644",
+              sha: null,
+            });
+            continue;
+          } catch (error) {
+            // istanbul ignore next
```

Comment:
I thought this is fine since I saw basically the same for other catch blocks, but please let me know if you would prefer a test case for this.